src/Services/CreditCards/ICreditCardsServices.cs

```diff
@@ -0,0 +1,151 @@
+﻿using CloudDrive.Domain;
+using CloudDrive.Domain.Entities;
+using CloudDrive.Persistence;
+using Microsoft.EntityFrameworkCore;
+using Services.CreditCards;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CloudDrive.Services.CreditCards
+{
+    public interface ICreditCardsServices
+    {
+        Task<Result<CreditCardsDto>> GetUserCreditCardAsync(int id);
+        Task<Result<List<CreditCardsDto>>> GetUserCreditCardsForUserAsync(int id);
+        Task<Result<CreditCardsDto>> CreateUserCreditCard(UserCreditCard userCreditCard);
+        Task<Result> DeleteUserCreditCardAsync(int id);
+        Task<Result<CreditCardsDto>> UpdateUserCreditCardAsync(int id, UserCreditCard userCreditCard);
+    }
+
+    public class UserCreditCardsImpl : ICreditCardsServices
+    {
+        //Connect the Database 
+        private readonly AppDbContext _appDbContext;
+        public UserCreditCardsImpl(AppDbContext appDbContext)
+        {
+            _appDbContext = appDbContext;
+        }
+
+
+        public async Task<Result<CreditCardsDto>> CreateUserCreditCard(UserCreditCard userCreditCard)
+        {
+            _appDbContext.UserCreditCards.Add(userCreditCard);
+            await _appDbContext.SaveChangesAsync();
+            return new Result<CreditCardsDto>
+            {
+                IsSuccssfull = true,
+                Data = new CreditCardsDto
+                {
+                    HolderName = userCreditCard.HolderName,
+                    CreditCardNumber = userCreditCard.CreditCardNumber,
+                    CardSecretCode = userCreditCard.CreditCardSecretCode,
+                    ExpireMonth = userCreditCard.ExpireMonth,
+                    ExpireYear = userCreditCard.ExpireYear,
+                }
+            };
+
+        }
+
+        public async Task<Result> DeleteUserCreditCardAsync(int id)
+        {
+            var CardToDelete = _appDbContext.UserCreditCards.Where(x=>x.Id == id).FirstOrDefault();
```

Comment:
You can use FirstOrDefault(x => x.Id == id) directly, but the best way it to use Find(id) as it uses the pk without searching thorugh the entire db